From: Simon McVittie <smcv@debian.org>
Date: Thu, 21 Dec 2017 14:31:49 +0000
Subject: gdbus-peer: Skip test during Debian package build

This test exposes a bug (Debian #884654, GNOME #791754) which does not
seem sufficiently serious to be release-critical. Skip it during the
build to avoid FTBFS, but continue to run it as an autopkgtest.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=791754
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884654
Forwarded: not-needed, Debian-specific
---
 gio/tests/gdbus-peer.c | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/gio/tests/gdbus-peer.c b/gio/tests/gdbus-peer.c
index b4169ab..f9531fe 100644
--- a/gio/tests/gdbus-peer.c
+++ b/gio/tests/gdbus-peer.c
@@ -669,6 +669,12 @@ test_peer (void)
   GThread *service_thread;
   gulong signal_handler_id;
 
+  if (g_getenv ("DEB_BUILD_TIME_TESTS") != NULL)
+    {
+      g_test_skip ("https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884654");
+      return;
+    }
+
   memset (&data, '\0', sizeof (PeerData));
   data.current_connections = g_ptr_array_new_with_free_func (g_object_unref);
 
@@ -1115,6 +1121,12 @@ delayed_message_processing (void)
   GThread *service_thread;
   guint n;
 
+  if (g_getenv ("DEB_BUILD_TIME_TESTS") != NULL)
+    {
+      g_test_skip ("https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884654");
+      return;
+    }
+
   data = g_new0 (DmpData, 1);
 
   service_thread = g_thread_new ("dmp",
@@ -1257,6 +1269,12 @@ test_nonce_tcp (void)
   gboolean res;
   const gchar *address;
 
+  if (g_getenv ("DEB_BUILD_TIME_TESTS") != NULL)
+    {
+      g_test_skip ("https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884654");
+      return;
+    }
+
   memset (&data, '\0', sizeof (PeerData));
   data.current_connections = g_ptr_array_new_with_free_func (g_object_unref);
 
@@ -1367,6 +1385,12 @@ test_credentials (void)
   GError *error;
   gchar *desc;
 
+  if (g_getenv ("DEB_BUILD_TIME_TESTS") != NULL)
+    {
+      g_test_skip ("https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884654");
+      return;
+    }
+
   c1 = g_credentials_new ();
   c2 = g_credentials_new ();
 
@@ -1442,6 +1466,12 @@ test_tcp_anonymous (void)
   GDBusConnection *connection;
   GError *error;
 
+  if (g_getenv ("DEB_BUILD_TIME_TESTS") != NULL)
+    {
+      g_test_skip ("https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884654");
+      return;
+    }
+
   seen_connection = FALSE;
   service_thread = g_thread_new ("tcp-anon-service",
                                  tcp_anonymous_service_thread_func,
@@ -1613,6 +1643,12 @@ codegen_test_peer (void)
   GVariant            *value;
   const gchar         *s;
 
+  if (g_getenv ("DEB_BUILD_TIME_TESTS") != NULL)
+    {
+      g_test_skip ("https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884654");
+      return;
+    }
+
   /* bring up a server - we run the server in a different thread to avoid deadlocks */
   service_thread = g_thread_new ("codegen_test_peer",
                                  codegen_service_thread_func,
